docs/_docs/01_getting-started/terminology.md

```diff
@@ -0,0 +1,209 @@
+
+This graph is also used to determine the order in which resources are resolved within a unit. Dependencies in a DAG determine the order in which resources are created, updated, or destroyed.
+
+For creations and updates, resources are updated such that dependencies are always resolved before their dependents. For destructions, resources are destroyed such that dependents are always destroyed before their dependencies.
+
+This is still true even when working with multiple units in a stack. Terragrunt will resolve the dependencies of all units in a stack (resolving the DAG within each unit first), and then apply the changes to all units in the stack in the correct order.
+
+Note that DAGs are _Acyclic_, meaning that there are no loops in the graph. This is because loops would create circular dependencies, which would make it impossible to determine the correct order to resolve resources.
+
+### Run
+
+A run is a single invocation of OpenTofu/Terraform by Terragrunt.
+
+Runs are the primary way that Terragrunt does work. When you run `terragrunt plan` or `terragrunt apply`, Terragrunt will invoke OpenTofu/Terraform to drive the infrastructure update accordingly.
+
+Note that runs abstract away a lot of the complexity that comes from working with OpenTofu/Terraform directly. Terragrunt might automatically perform some code generation, provision requisite resources, or add/modify to the underlying OpenTofu/Terraform configuration to ensure that day to day operations are as smooth as possible.
+
+The way in which these complexities are abstracted is via Terragrunt configuration files (`terragrunt.hcl`), which can be used to define how Terragrunt should forward commands to OpenTofu/Terraform.
+
+In the simplest case, a run in a unit with an empty `terragrunt.hcl` file will be equivalent to running OpenTofu/Terraform directly in the unit directory (with some small additional features like automatic initialization and logging adjustments).
+
+### Execution
+
+An execution is a single command run by Terragrunt, which does not necessarily have anything to do with OpenTofu/Terraform.
+
+Ways in which Terragrunt can perform executions are limited to features like [hooks](/docs/features/hooks/), [run_cmd](/docs/reference/built-in-functions/#run_cmd), etc.
+
+These utilities are part of what makes Terragrunt so powerful, as they allow users to move infrastructure management complexity out of modules.
+
+### Runner Queue
+
+The Runner Queue is the queue of all units that Terragrunt will do work on over one or more runs.
+
+Certain commands like [run-all](/docs/reference/cli-options/#run-all) ([soon to be rebranded `run --all`](https://github.com/gruntwork-io/terragrunt/issues/3445)) populate the Runner Queue with all units in a stack, while other commands like `plan` or `apply` will only populate the Runner Queue with the unit that the command was run in.
```

Comment:
I think the soon to be rebranded comments might be better as  footnotes at the end of each section?